src/server/items/models.js

```diff
@@ -43,5 +43,33 @@ export default class ItemManager {
     }
     return (new Date().getTime() - new Date(endTime).getTime() < 0) ? true : false;
   }
+  static fillIsSaved(userKey, items) {
+    return UserManager.getPostKeys(ENTITY.SAVED_POST, userKey)
+    .then((posts)=>{
+      return new Promise((resolve)=>{
+        posts.sort((a, b) => {
+          return a.key > b.key;
+        });
+        const itemKeys = this.getFields(items, 'key');
+        const postKeys = this.getFields(posts, 'key');
+        let pointer = 0;
+        for (let i = 0; i < itemKeys.length; i = i + 1) {
+          const idx = postKeys.indexOf(itemKeys[i], pointer);
+          if (idx !== -1) {
```

Comment:
It means there is no itemKeys[i] value in postKeys[pointer~its length]~~.
if it has value, idx is index of value in postKeys.